spec/features/advanced_search_spec.rb

```diff
@@ -0,0 +1,176 @@
+          click_button 'advanced-search-submit'
+        end
+
+        it 'results include expected CAT keys' do
+          expect(page).to have_selector 'article[data-document-id="31805602"]'
+        end
+      end
+
+      context 'when searching by publication date range' do
+        before do
+          fill_in 'range_pub_date_itsi_begin', with: '2017'
+          fill_in 'range_pub_date_itsi_end', with: '2018'
+          click_button 'advanced-search-submit'
+        end
+
+        it 'results include expected CAT keys' do
+          expect(page).to have_selector 'article[data-document-id="31805602"]'
+          expect(page).to have_selector 'article[data-document-id="31805615"]'
+        end
+      end
+    end
+
+    context 'when doing combination searches' do
+      context 'when searching for a specific book about quilts in Cumberland County' do
+        before do
+          fill_in 'title', with: 'quilts cumberland'
+          fill_in 'author', with: 'quilters'
+          find('button[data-id="access_facet"]').click
+          find('#bs-select-1-1').click
+          find('button[data-id="format"]').click
+          find('#bs-select-2-3').click
+          find('button[data-id="language_facet"]').click
+          find('#bs-select-3-4').click
+          find('button[data-id="lc_1letter_facet"]').click
+          find('#bs-select-5-12').click
+          click_button 'advanced-search-submit'
+        end
+
+        it 'results include expected CAT keys' do
+          expect(page).to have_selector 'article[data-document-id="18648055"]'
+        end
+      end
+
+      context 'when searching for a specific book on geological perspectives of climate change' do
+        before do
+          fill_in 'subject', with: 'Climatology'
+          fill_in 'series', with: 'AAPG studies in geology'
+          fill_in 'publisher', with: 'Petroleum Geologists'
+          find('button[data-id="library_facet"]').click
+          find('#bs-select-6-4').click
+          click_button 'advanced-search-submit'
+        end
+
+        it 'results include expected CAT keys' do
+          expect(page).to have_selector 'article[data-document-id="2250425"]'
+        end
+      end
+    end
+
+    context 'when using vernacular language' do
+      context 'when searching in simplified chinese' do
+        before do
+          fill_in 'title', with: '上海理论'
+          click_button 'advanced-search-submit'
+        end
+
+        it 'results include expected CAT keys' do
+          expect(page).to have_selector 'article[data-document-id="17280643"]'
+        end
+      end
+
+      context 'when searching in traditional chinese' do
+        before do
+          fill_in 'title', with: '上海理論'
+          click_button 'advanced-search-submit'
+        end
+
+        it 'results include expected CAT keys' do
+          expect(page).to have_selector 'article[data-document-id="17280643"]'
+        end
+      end
+    end
+
+    # should this be in another location?
```

Comment:
It may be a good idea to move this somewhere else since it isn't really related to advanced search.  Would it be helpful to also test clicking the link and testing that the marc record shows?  Not sure if that's being tested anywhere already.

Reply:
agree with Alex, thanks for your review, good point! and thanks Sarah for refactoring that part.